File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerChatEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Objects;

public final class PlayerChatEvent extends AbstractCancellableEvent {

    private final CellPlayer player;
    private String message;

    public PlayerChatEvent(
            CellPlayer player,
            String message
    ) {
        this.player = Objects.requireNonNull(player, "player");
        this.message = Objects.requireNonNull(message, "message");
    }

    public CellPlayer player() {
        return player;
    }

    public String message() {
        return message;
    }

    public void message(String message) {
        this.message = Objects.requireNonNull(message, "message");
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerChatEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer

public class PlayerChatEvent(
    private val player: CellPlayer,
    private var message: String
) : AbstractCancellableEvent() {

    public fun player(): CellPlayer = player

    public fun message(): String = message

    public fun message(message: String) {
        this.message = message
    }

}
